File: Linked Lists/1290. Convert Binary Number in a Linked List to Integer.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def getDecimalValue(self, head: Optional[ListNode]) -> int:
        decimal = 0
        curr = head
        while curr:
            decimal = (decimal << 1) | curr.val
            curr = curr.next
        return decimal

# Time: O(n)
# Space: O(1)

File: Linked Lists/1290. Convert Binary Number in a Linked List to Integer/1290. Convert Binary Number in a Linked List to Integer.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {number}
 */
var getDecimalValue = function (head) {
  let decimal = 0;
  let curr = head;
  while (curr) {
    decimal = (decimal << 1) + curr.val;
    curr = curr.next;
  }
  return decimal;
};

// Time: O(n)
// Space: O(1)
